protostar/cheatable_starknet/controllers/expect_events_controller.py

```diff
@@ -0,0 +1,166 @@
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Optional
+
+from protostar.starknet import Address, CairoData, Selector, ReportedException
+
+if TYPE_CHECKING:
+    from protostar.testing import Hook
+    from protostar.cairo_testing import CairoTestExecutionState
+    from protostar.cheatable_starknet.cheatables.cheatable_cached_state import (
+        CheatableCachedState,
+    )
+
+
+@dataclass
+class Event:
+    from_address: Address
+    key: Selector
+    data: Optional[CairoData] = None
+
+
+@dataclass
+class EventMatching:
+    pass
+
+
+@dataclass
+class SkippedEventMatching(EventMatching):
+    emitted_event: Event
```

Comment:
Do we need to acknowledge they are even present? The only thing we're interested in are failed matchings (meaning ones that are expected, or not there, or mismatched messages)

Reply:
We need to tell the user what events were actually emitted. Without that, it's hard to say what went wrong.